cpp/src/arrow/compute/exec/exec_plan.cc

```diff
@@ -105,18 +107,11 @@ struct ExecPlanImpl : public ExecPlan {
           return Status::OK();
         }
 
-        auto it_success = visiting.insert(node);
-        if (!it_success.second) {
-          // Insertion failed => node is already being visited
-          return Status::Invalid("Cycle detected in execution plan");
-        }
```

Comment:
Why did you remove this?

Reply:
Since nodes are now constructed with their inputs it's impossible to produce a cycle